lib/services/local/integrations.go

```diff
@@ -173,6 +197,41 @@ func notReferencedByEAS(ctx context.Context, bk backend.Backend, name string) ([
 	return conditionalActions, nil
 }
 
+// integrationReferencedByAWSICPlugin returns an error if the integration name is referenced
+// by an existing AWS Identity Center plugin. In case the AWS Identity Center plugin exists
+// but does not reference this integration, a conditional action is returned with a revision
+// of the plugin to ensure that plugin hasn't changed during deletion of the AWS OIDC integration.
+// Returns nil ConditionalAction if AWS Identity Center plugin is not found.
+func integrationReferencedByAWSICPlugin(ctx context.Context, bk backend.Backend, name string) (*backend.ConditionalAction, error) {
```

Comment:
The `integrationReferencedByEAS` function  always returns conditional action but this `integrationReferencedByAWSICPlugin` does not. I do not think we need to add condition for a non-existent plugin (feasibility is also questionable as there is no point on referencing another plugin which is not AWS Identity Center) but I am open to suggestion otherwise.